datatree/mapping.py

```diff
@@ -212,6 +212,8 @@ def _map_over_subtree(*args, **kwargs) -> DataTree | Tuple[DataTree, ...]:
                     *node_args_as_datasetviews, **node_kwargs_as_datasetviews
                 )
                 if node_of_first_tree.has_data
+                else node_of_first_tree.ds
+                if node_of_first_tree.has_attrs
                 else None
```

Comment:
I did not even know that this double `if else` was valid python!

Could we perhaps rewrite this to be more intuitive, e.g.

```python
if node.has_data:
    result = func_with_error_context(...)
elif node.has_attrs:
    # propagate attrs
    result = node.ds
else:
    # nothing to propagate so use fastpath to create empty node in new tree
    result = None
```

Reply:
lol, neither did I. For some reason when I first looked at these lines I thought it was a list comprehension and had to have google remind me whether you can `elif` in a comprehension (you can not, but you can do this weird syntax :shrug: ). Fixed.